### PR TITLE
fix(android): fix ListView layout with header and footer only

### PR DESCRIPTION
**Pull request by m1ga**
There is an issue with ListView sections that don't have a content and only a header and footer:

Screenshot_20240917-211111

```xml
<Alloy>
	<Window>
		<ListView>
			<ListSection>
				<ListItem title="Item" />
				<ListItem title="Item" />
			</ListSection>

			<ListSection>
				<HeaderView>
					<View height="50" backgroundColor="#111">
						<Label>Section 2 : Header View - item height=0</Label>
					</View>
				</HeaderView>

				<ListItem height="0"/>

				<FooterView>
					<View height="25" backgroundColor="#444">
						<Label>Section 2 : Footer View</Label>
					</View>
				</FooterView>
			</ListSection>

			<ListSection>
				<HeaderView>
					<View height="50" backgroundColor="#111">
						<Label>Section 2 : Header View - no item</Label>
					</View>
				</HeaderView>

				<FooterView>
					<View height="25" backgroundColor="#444">
						<Label>Section 2 : Footer View</Label>
					</View>
				</FooterView>
			</ListSection>

			<ListSection>
				<FooterView>
					<View height="25" backgroundColor="#444">
            <Label>Section only footer View - no item</Label>
					</View>
				</FooterView>
			</ListSection>

			<ListSection>
        <ListItem title="Item from 4" />
        <ListItem title="Item from 4" />
				<FooterView>
					<View height="25" backgroundColor="#444">
            <Label>Section 4 only footer View - and items</Label>
					</View>
				</FooterView>
			</ListSection>

			<ListSection>
				<ListItem title="last Section Item 1" />
				<ListItem title="last Section Item 2" />
				<ListItem title="last Section Item 3" />
			</ListSection>
		</ListView>
	</Window>
</Alloy>
```

This PR will connect the footer top to the header bottom if there isn't any content:
Screenshot_20240917-210822

Current workaround: add `<ListItem height="0"/>`

### CODE REVIEW

**Comment by hansemannn** [comment]
cc @prashantsaini1 can you review?

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/ListViewHolder.java [hunk 1]
```diff
@@ -412,8 +412,15 @@ private void setHeaderFooter(TiViewProxy listViewProxy,
...
-							R.id.titanium_ui_listview_holder_footer, ConstraintSet.TOP, 0);
+						if (this.content.getChildCount() > 0) {
+							// attach to content
+							constraintSet.connect(R.id.titanium_ui_listview_holder_content, ConstraintSet.BOTTOM,
+								R.id.titanium_ui_listview_holder_footer, ConstraintSet.TOP, 0);
+						} else {
+							// attach to header
+							constraintSet.connect(R.id.titanium_ui_listview_holder_header, ConstraintSet.BOTTOM,
+								R.id.titanium_ui_listview_holder_footer, ConstraintSet.TOP, 0);
+						}
```
Comment (cb1kenobi) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/ListViewHolder.java:423]: Generally when you have a duplicate function call with many shared arguments, it's easier to read and maintain if you set the thing that changed in its own variable like this:

```suggestion

						int holderId = R.id.titanium_ui_listview_holder_header;
						if (this.content.getChildCount() > 0) {
							holderId = R.id.titanium_ui_listview_holder_content;
						}
						
						constraintSet.connect(holderId, ConstraintSet.BOTTOM,
							R.id.titanium_ui_listview_holder_footer, ConstraintSet.TOP, 0);
```

Also, `holderId` is a horrible name. The argument is `startId`.

**Review by cb1kenobi — APPROVED** [review]
Tested on Windows. Fix works as advertised. Thanks!

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/ListViewHolder.java (hunk 1 above)
Comment (m1ga) [reply android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/ListViewHolder.java:423]: I'll change it to `constraintStartId` but I like this way

Comment (hansemannn) [reply android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/ListViewHolder.java]: @cb1kenobi "holder" is a actually a common terminology in the Android world for these cases, especially with [`RecyclerView` instances](https://developer.android.com/reference/androidx/recyclerview/widget/RecyclerView.ViewHolder). I would keep it if it's okay for you :)

**Review by cb1kenobi — APPROVED** [review]

**Review by hansemannn — APPROVED** [review]

**Comment by prashantsaini1** [comment]
I noticed a major issue in recent changes regarding layout updates. I raised a PR to fix it properly. Hopefully we can test it there.

**Pull request merged**